compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -959,25 +965,49 @@ execMergeDars darFp1 darFp2 mbOutFp =
         pure $ ZipArchive.toEntry manifestPath 0 $ BSLC.unlines $
             map (\(k, v) -> breakAt72Bytes $ BSL.fromStrict $ k <> ": " <> v) attrs1
 
-execDocTest :: Options -> [FilePath] -> Command
-execDocTest opts files =
+execDocTest :: Options -> FilePath -> Bool -> [FilePath] -> Command
+execDocTest opts scriptDar noSource files =
   Command DocTest Nothing effect
   where
     effect = do
       let files' = map toNormalizedFilePath' files
+          packageFlag =
+            ExposePackage
+              ("--package daml-script-" <> SdkVersion.sdkPackageVersion)
+              (UnitIdArg $ stringToUnitId $ "daml-script-" <> SdkVersion.sdkPackageVersion)
+              (ModRenaming True [])
+
       logger <- getLogger opts "doctest"
+
+      -- Install daml-script in their project and update the package db
+      -- TODO: Should this be "."?
+      installDependencies "." opts (PackageSdkVersion SdkVersion.sdkVersion) [scriptDar] []
+      createProjectPackageDb "." opts mempty
```

Comment:
Unsure what to do here - we _could_ add `projectOpts` to the `damlc doctest` command, which would give us a path to work from, but that might be a breaking change.
We could consider a temp directory, but then we'd need to copy over any installed modules there, alongside source files.
"." works, but isn't very safe, creating a `.daml` file in whatever folder you call this command from, as `build` would.

Reply:
as discussed, this would add `daml-script` to the package db even if the project itself doesn't use it, so it (might?) end up in the project dar by accident. it's not great, but considering 1. `daml doctest` is in alpha/early access, 2. it doesn't seem to have many users and 3. most projects do use `daml-script` anyway, it's not worth it putting much effort into improving this now. For now I'd be happy if the comment above pointed to this PR for context.